specification/imds/data-plane/Microsoft.InstanceMetadataService/stable/2018-10-01/imds.json

```diff
@@ -0,0 +1,731 @@
+{   
+    "swagger": "2.0",
+    "info": {
+        "title": "InstanceMetadataClient",
+        "description": "The Azure Instance Metadata Client",
+        "version": "2018-10-01"
+    },
+    "host": "169.254.169.254",
+    "basePath": "/metadata",
+    "schemes": [
+        "https"
+    ],
+    "consumes": [
+        "application/json"
+    ],
+    "produces": [
+        "application/json"
+    ],
+    "security": [
+        {
+            "azure_auth": [
+                "user_impersonation"
+            ]
+        }
+    ],
+    "securityDefinitions": {
+        "azure_auth": {
+            "type": "oauth2",
+            "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+            "flow": "implicit",
+            "description": "Azure Active Directory OAuth2 Flow",
+            "scopes": {
+                "user_impersonation": "impersonate your user account"
+            }
+        }
+    },
+    "paths": {
+        "/instance": {
+            "get": {
+                "operationId": "getInstance",
```

Comment:
We recommend operationId to be in the "Noun_Verb" format. This makes it easier for developers using the sdk to find logical operation groups on the client and then methods/actions that can be performed on those logical entities. Currently you would see all the methods on the client. This provides a subpar developer experience.

Reply:
operationId should be "Instances_Get". The reason to make the Noun plural is that "there is already a model definition named "Instance". Pluralizing operation groups resolves the naming conflict.

Reply:
Operations have been renamed to Instance_GetMetadata, Attested_GetDocument, Identity_GetToken, and Identity_GetInfo.